src/property-filter/__tests__/property-filter-token-editor.test.tsx

```diff
@@ -0,0 +1,310 @@
+    groupValuesLabel: 'Other string values',
+  },
+  // operator unspecified
+  {
+    key: 'default-operator',
+    propertyLabel: 'default',
+    groupValuesLabel: 'Default values',
+  },
+  // supports range operators
+  {
+    key: 'range',
+    propertyLabel: 'range',
+    operators: ['>', '<', '=', '!=', '>=', '<='],
+    groupValuesLabel: 'Range values',
+    group: 'group-name',
+  },
+  // property label is consists of another property with an operator after
+  {
+    key: 'edge-case',
+    propertyLabel: 'string!=',
+    operators: ['!:', ':', '=', '!='],
+    groupValuesLabel: 'Edge case values',
+  },
+  {
+    key: 'state',
+    propertyLabel: 'state',
+    operators: [{ operator: '=', format: getStateLabel }],
+    groupValuesLabel: 'State values',
+  },
+];
+
+const filteringOptions: readonly FilteringOption[] = [
+  { propertyKey: 'string', value: 'value1' },
+  { propertyKey: 'other-string', value: 'value1' },
+  { propertyKey: 'string', value: 'value2' },
+  { propertyKey: 'range', value: '1' },
+  { propertyKey: 'range', value: '2' },
+  { propertyKey: 'other-string', value: 'value2' },
+  { propertyKey: 'missing-property', value: 'value' },
+  { propertyKey: 'default-operator', value: 'value' },
+  { propertyKey: 'state', value: '0', label: getStateLabel('0') },
+  { propertyKey: 'state', value: '1', label: getStateLabel('1') },
+  { propertyKey: 'state', value: '2', label: getStateLabel('2') },
+];
+
+const defaultProps = createDefaultProps(filteringProperties, filteringOptions);
+
+const renderComponent = (props?: Partial<PropertyFilterProps & { ref: React.Ref<Ref> }>) => {
+  const { container } = render(<PropertyFilter {...defaultProps} {...props} />);
+  return { container, propertyFilterWrapper: createWrapper(container).findPropertyFilter()! };
+};
+
+function findPropertyField(wrapper: ElementWrapper) {
+  return wrapper.findFormField(`.${styles['token-editor-field-property']}`)!;
+}
+function findOperatorField(wrapper: ElementWrapper) {
+  return wrapper.findFormField(`.${styles['token-editor-field-operator']}`)!;
+}
+function findValueField(wrapper: ElementWrapper) {
+  return wrapper.findFormField(`.${styles['token-editor-field-value']}`)!;
+}
+
+function findPropertySelector(wrapper: ElementWrapper) {
+  return findPropertyField(wrapper).findControl()!.findSelect()!;
+}
+function findOperatorSelector(wrapper: ElementWrapper) {
+  return findOperatorField(wrapper).findControl()!.findSelect()!;
+}
+function findValueSelector(wrapper: ElementWrapper) {
+  return findValueField(wrapper).findControl()!.findAutosuggest()!;
+}
+
+function openTokenEditor(wrapper: PropertyFilterWrapper, index = 0) {
+  const tokenWrapper = createWrapper(wrapper.findTokens()![index].getElement());
+  const popoverWrapper = tokenWrapper.findPopover()!;
+  act(() => popoverWrapper.findTrigger().click());
+  const contentWrapper = popoverWrapper.findContent()!;
+  return [contentWrapper, popoverWrapper] as const;
+}
```

Comment:
_Scope: From the line `88` to `119`._

We've used some of the functions multiple times in different files. Would it make sense to move these definitions to the `common.tsx`.

Reply:
I plan not to do it just now because these common function better be the official public test-utils methods. I will work on the proposal.